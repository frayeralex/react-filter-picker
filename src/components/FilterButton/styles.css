.FilterButton {
  box-sizing: content-box;
  border: 1px solid #000;
  padding: 10px;
  margin: 0 5px;
  min-width: 60px;
  text-align: center;
  flex-grow: 1;
}

.FilterButton_active {
  background: rgba(0, 0,0, .1);
}
