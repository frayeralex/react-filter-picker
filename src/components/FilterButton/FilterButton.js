import React from 'react'
import classNames from 'classnames'
import styles from './styles.css'
import PropTypes from 'prop-types'

const FilterButton = (props) => {
  let boxRef = React.createRef();

  const handleOnClick = (event) => {
    event.stopPropagation();
    event.nativeEvent.stopImmediatePropagation();

    props.onClick({
      filter: props.filter,
      coordinates: boxRef.current.getBoundingClientRect(),
    });
  }

  return (
    <div
      ref={boxRef}
      onClick={handleOnClick}
      className={classNames(styles.FilterButton, {
        [styles.FilterButton_active]: props.active
      })}>
      <span className="label">{props.filter}</span>
      {props.selectedCount > 0 && <span className="selected-filters-count">{' '}({props.selectedCount})</span>}
    </div>
  )
}

FilterButton.propTypes = {
  filter: PropTypes.string.isRequired,
  selectedCount: PropTypes.number,
}

FilterButton.defaultProps = {
  selectedCount: 0,
}

export default FilterButton;
