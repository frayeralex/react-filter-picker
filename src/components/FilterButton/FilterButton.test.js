import React from 'react';
import FilterButton from './FilterButton';

describe('<FilterButton/>', () => {
  it('should render without crashing', () => {
    const wrapper = shallow(<FilterButton filter="color" />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should show selected count correctly', function() {
    const wrapper = shallow(<FilterButton filter="color" />);
    wrapper.setProps({ selectedCount: 20 });
    expect(
      wrapper
        .find('.selected-filters-count')
        .first()
        .text()
        .trim(),
    ).toEqual('(20)');
    expect(wrapper).toMatchSnapshot();
  });
  it('should handle click correct', function() {
    const onClick = jest.fn();
    const stopPropagation = jest.fn();
    const stopImmediatePropagation = jest.fn();
    const wrapper = mount(<FilterButton filter="color" onClick={onClick} />);
    wrapper.root().simulate('click', {
      stopPropagation,
      nativeEvent: {
        stopImmediatePropagation,
      },
    });
    expect(stopPropagation).toHaveBeenCalled();
    expect(stopImmediatePropagation).toHaveBeenCalled();
    expect(onClick).toHaveBeenCalledWith({
      filter: 'color',
      coordinates: expect.objectContaining({
        bottom: expect.any(Number),
        height: expect.any(Number),
        width: expect.any(Number),
        left: expect.any(Number),
        top: expect.any(Number),
        right: expect.any(Number),
      }),
    });
  });
});
