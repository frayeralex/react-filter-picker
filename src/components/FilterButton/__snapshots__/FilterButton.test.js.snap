// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilterButton/> should render without crashing 1`] = `
<div
  className=""
  onClick={[Function]}
>
  <span
    className="label"
  >
    color
  </span>
</div>
`;

exports[`<FilterButton/> should show selected count correctly 1`] = `
<div
  className=""
  onClick={[Function]}
>
  <span
    className="label"
  >
    color
  </span>
  <span
    className="selected-filters-count"
  >
     (
    20
    )
  </span>
</div>
`;
